src/primitives/hrp.rs

```diff
@@ -0,0 +1,420 @@
+// SPDX-License-Identifier: MIT
+
+//! Provides an `Hrp` type that represents the human-readable part of a bech32 encoded string.
+//!
+//! > The human-readable part, which is intended to convey the type of data, or anything else that
+//! > is relevant to the reader. This part MUST contain 1 to 83 US-ASCII characters, with each
+//! > character having a value in the range [33-126]. HRP validity may be further restricted by
+//! > specific applications.
+//!
+//! ref: [BIP-173](https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki#user-content-Bech32)
+
+#[cfg(feature = "alloc")]
+use alloc::string::String;
+use core::cmp::Ordering;
+use core::fmt::{self, Write};
+use core::iter::FusedIterator;
+use core::slice::Iter;
```

Comment:
In 599ce6d089904386ac6a34d77a6b3b1fe222393f:

I think it's confusing to import this type. I'd import `slice` and refer to it as `slice::Iter` instead.

Reply:
Ah yes, that is nicer.